app_dart/micro_services/results_pipeline/bin/server.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:appengine/appengine.dart';
+import 'package:results_pipeline/common/auth_utils.dart';
+import 'package:results_pipeline/common/logging.dart';
+import 'package:results_pipeline/common/response.dart';
+
+/// Default handler, returns a 404
+void _defaultHandler(HttpRequest request) {
+  respond(request, status: HttpStatus.notFound);
+}
+
+/// Check authorization of incoming request.
+Future<void> _checkAuthorization(HttpRequest request) async {
+  if (!await authenticateRequest(request.headers)) {
+    await respond(request, status: HttpStatus.unauthorized);
+  }
+}
+
+/// Processes LUCI build status updates messages.
+///
+/// Endpoint for push messages from the pubsub topic that
+/// LUCI send build update status messages to.
+/// Tasks:
+///  1) Verify authorization on incoming message.
+///  2) Decodes build update message.
+///  3) Saves update to datatstore.
+///  4) Triggers retry for failing build.
+///
+/// POST: /api/process_build_request
+///
+/// Response: Status 200 OK
+///
+/// Response: Status 401 Unauthorized
+///
+Future<void> _processUpdate(HttpRequest request) async {
+  await _checkAuthorization(request);
```

Comment:
@godofredoc also, I was thinking of moving the auth checks out to _requestHandler(), rather than make it the responsibility of each handler. It might help avoid times where someone forgets to add necessary auth to a handler, and cosolidates the checks at the top level. Thougts?

Reply:
The reason for auth per request handler is it enables ACLs. Otherwise, we have one generic auth for the entire backend

Reply:
I didn't mean have one for every handler, but have the calls made just above the handler call each time. E.g.:

Future<void> _requestHandler(HttpRequest request) async {
  if (request.uri.path == '/api/process_build_update') {
    checkAUTH here;
    await _processUpdate(request);
  } else {
   checkAUTH here;
    _defaultHandler(request);
  }
}

This separates responsibility from the handler to do one additional function, simplifying it.